nkms/client.py

```diff
@@ -27,6 +30,26 @@ def __init__(self, conf=None):
             or created
         """
         self._nclient = Client.network_client_factory()
+        self._pre = pre_from_algorithm(default_algorithm)
+        self._symm = symmetric_from_algorithm(default_algorithm)
+
+        # TODO: Check for existing keypair before generation
+        # TODO: Save newly generated keypair
+        self._priv_key = self._pre.gen_priv(dtype='bytes')
+        self._pub_key = self._pre.priv2pub(self._priv_key)
+
+    def _derive_path_key(self, path, is_pub=True):
+        """
+        Derives a public key for the specific path.
+
+        :param str path: Path to generate key for.
```

Comment:
Does it make more sense for path to be `bytes`?  Some users might have path names that are not utf-8.

Reply:
I think it does indeed!

Reply:
Very good point. Thanks!